major_event_log/feeds.py

```diff
@@ -8,7 +8,54 @@
 from .models import Event
 
 
-class MajorEventLogFeed(Atom1Feed):
+class PaginatedFeedTypeMixin(object):
+
+    def _create_link_attr(self, rel, page):
+        href = u'{0}?{1}={2}'.format(
+            self.feed['link'], self.feed['page_query'], page)
+
+        return {u'rel': rel, u'href': href}
+
+    def add_root_elements(self, handler):
+        from django.utils.feedgenerator import rfc3339_date
+        handler.addQuickElement("title", self.feed['title'])
+        handler.addQuickElement(
+            "link", "", {"rel": "alternate", "href": self.feed['link']})
+
+        if self.feed['feed_url'] is not None:
+            handler.addQuickElement("link", "", {"rel": "self", "href": self.feed['feed_url']})
+
+        handler.addQuickElement(u'link', '', self._create_link_attr(u'first', 1))
+        handler.addQuickElement(u'link', '', self._create_link_attr(u'last', self.feed['last_page']))
+
+        if self.feed.get('prev_page', None) is not None:
+            handler.addQuickElement(u'link', '', self._create_link_attr(u'prev_page', self.feed['prev_page']))
+        if self.feed.get('next_page', None) is not None:
+            handler.addQuickElement(u'link', '', self._create_link_attr(u'next_page', self.feed['next_page']))
+
+        handler.addQuickElement("id", self.feed['id'])
+        handler.addQuickElement("updated", rfc3339_date(self.latest_post_date()))
+
+        if self.feed['author_name'] is not None:
+            handler.startElement("author", {})
+            handler.addQuickElement("name", self.feed['author_name'])
+            if self.feed['author_email'] is not None:
+                handler.addQuickElement("email", self.feed['author_email'])
+            if self.feed['author_link'] is not None:
+                handler.addQuickElement("uri", self.feed['author_link'])
+            handler.endElement("author")
+
+        if self.feed['subtitle'] is not None:
+            handler.addQuickElement("subtitle", self.feed['subtitle'])
+
+        for cat in self.feed['categories']:
+            handler.addQuickElement("category", "", {"term": cat})
```

Comment:
The reason is because that way we maintain the original functionality of `add_root_elements()`, but can customize the location of where the paginated links appear.